source/initial_conditions/adiabatic.cc

```diff
@@ -58,6 +52,21 @@ namespace aspect
                                                   :
                                                   adiabatic_surface_temperature;
 
+      // then, get the temperature at the top and bottom boundary of the model
+      // if no boundary temperature is prescribed simply use the adiabatic
+      const double T_surface = (&this->get_boundary_temperature() != 0)
+                               ?
+                                   this->get_boundary_temperature().minimal_temperature(
+                                       this->get_fixed_temperature_boundary_indicators())
+                               :
+                                   adiabatic_surface_temperature;
+      const double T_bottom = (&this->get_boundary_temperature() != 0)
+                              ?
+                                  this->get_boundary_temperature().maximal_temperature(
+                                      this->get_fixed_temperature_boundary_indicators())
+                              :
+                                  adiabatic_surface_temperature;
+
```

Comment:
Ok, now I understand why we chose to do it like this. Boundary temperature needs the boundary_id of the point for which you would like to calculate the temperature, and there is no way to know in general, which boundary is the top and bottom (they are different for every geometry_model). For now it is easier to update the comment.